android/KMEA/app/src/main/java/com/tavultesoft/kmea/JSONParser.java

```diff
@@ -19,16 +22,71 @@
 
 import android.util.Log;
 
-final class JSONParser {
-
-  private static InputStream inputStream = null;
-  private static JSONObject jsonObj = null;
-  private static String jsonStr = "";
+public final class JSONParser {
 
   public JSONParser() {
   }
 
+  private JSONObject getJSONObjectFromReader(BufferedReader reader) {
+    String jsonStr = "";
+    JSONObject jsonObj = null;
+
+    try {
+      StringBuilder strBuilder = new StringBuilder();
+      String line = null;
+      while ((line = reader.readLine()) != null) {
+        strBuilder.append(line + "\n");
+      }
+      jsonStr = strBuilder.toString();
+      jsonObj = new JSONObject(jsonStr);
+    } catch (UnsupportedEncodingException e) {
+      Log.e("Encoding Error", (e.getMessage() == null) ? "UnsupportedEncodingException" : e.getMessage());
```

Comment:
Can we use `getJSONObjectFromReader` or `getJSONObjectFromFile` for the TAG portion of the `Log.e(TAG, ...)`?